V2rayNG/app/src/main/kotlin/com/v2ray/ang/util/AngConfigManager.kt

```diff
@@ -378,6 +378,60 @@ object AngConfigManager {
         return 0
     }
 
+    fun tryParseNewVmess(uri: String): AngConfig.VmessBean? {
+        return runCatching {
+            val uri = URI(uri)
+            check(uri.scheme == "vmess")
+            val (_, protocol, tlsStr, uuid, alterId) =
+                    Regex("(tcp|http|ws|kcp|quic)(\\+tls)?:([0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12})-([0-9]+)")
+                            .matchEntire(uri.userInfo)?.groupValues
+                            ?: error("parse user info fail.")
+            val tls = tlsStr.isNotBlank()
+            val queryParam = uri.rawQuery.split("&")
+                    .map { it.split("=").let { (k, v) -> k to URLDecoder.decode(v, "utf-8")!! } }
+                    .toMap()
+            val vmess = AngConfig.VmessBean()
+            vmess.address = uri.host
+            vmess.port = uri.port
+            vmess.guid = uuid
+            vmess.alterId = alterId.toInt()
+            vmess.streamSecurity = if (tls) "tls" else ""
+            vmess.remarks = uri.fragment
+
+            // TODO: allowInsecure not supported
+
+            when (protocol) {
+                "tcp" -> {
+                    vmess.network = "tcp"
+                    vmess.headerType = queryParam["type"] ?: "none"
+                    vmess.requestHost = queryParam["host"] ?: ""
+                }
+                "http" -> {
+                    vmess.network = "h2"
+                    vmess.path = queryParam["path"]?.takeIf { it.trim() != "/" } ?: ""
+                    vmess.requestHost = queryParam["host"]?.split("|")?.get(0) ?: ""
+                }
+                "ws" -> {
+                    vmess.network = "ws"
+                    vmess.path = queryParam["path"]?.takeIf { it.trim() != "/" } ?: ""
+                    vmess.requestHost = queryParam["host"]?.split("|")?.get(0) ?: ""
+                }
+                "kcp" -> {
+                    vmess.network = "kcp"
+                    vmess.headerType = queryParam["type"] ?: "none"
+                    vmess.path = queryParam["seed"] ?: ""
+                }
+                "quic" -> {
+                    vmess.network = "quic"
+                    vmess.security = queryParam["security"] ?: "none"
```

Comment:
Currently quic security should be put as vmess.requestHost.
This brings a question.. how does the new format carry vmess.securiity?

Reply:
现阶段，vmess的security取消，默认auto

Reply:
Is the new format finalised? If not I'd suggest we should add security.

Reply:
Until now, looks there is no plan for it.  And I don't think we should add this feature. Because we had tls already.

Reply:
no matter which encryption has been choosed, client can always connect to server, so I don't think security option is necessary.

Reply:
Hello.
As far as I could tell, VMess could perform auto-negotiation. So this is not required anymore.

![image](https://user-images.githubusercontent.com/59914293/88359102-0460ba00-cda4-11ea-98a0-d9b9452bf01a.png)

Reply:
Ok, it is not mandatory but it would be a convenient setting. "none" is a common choice with tls. Also user would expect to transfer basic settings with export-and-then-import. For instance..
https://github.com/2dust/v2rayNG/issues/356
dalao @rprx was surprised by it :)

Reply:
That only happens when using the old standard. It's your parser's fault not to work correctly.
The new standard obsoletes this setting, so they are irrelevant.

Reply:
The new version scheme has removed the paramter. I think it's viable to user, because of we don't use base64. So that user will understand what's happeded.